packages/react-theme-provider/src/ThemeProvider.stories.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { ThemeProvider } from './ThemeProvider';
+import { Theme } from './types';
+
+// tslint:disable:jsx-no-lambda
+
+const lightTheme: Theme = {
+  tokens: {
+    body: {
+      fill: 'white',
+      text: 'black',
+      fontFamily: 'Segoe UI',
+    },
+  },
+};
+
+const darkTheme: Theme = {
+  tokens: {
+    body: {
+      fill: 'black',
+      text: 'white',
+    },
+  },
+};
+
+const themeWithStylesheets: Theme = {
+  stylesheets: ['.foo { font-family: var(--body-customFont); }'],
```

Comment:
Is any validation performed on this API? It seems easy to provide an incorrect selector rule set.

Reply:
Certainly not at runtime, which would be a perf hit. Expectation is at build time. This is similar to `loadStyles`, with the exception that styles would be deferred to be registered until needed rather than on module load. This means

* less css rendering at module load time :+1:
* ability to project rules to child window :+1:
* load order will not be determined by module import order, but rather on component render order :-1:(but selectors should not depend on load order to resolve specificity; that's fragile)

```tsx
// Up to the css compiler build step to fail if you have bad css.
import { stylesheet: fileStyles } from './file.scss';

const theme: Theme = {
  stylesheets: [ fileStyles ]
};
```

Reply:
What happens if a developer defines `stylesheets: ['foo { bad-selector` ]`? Is the stylesheet appended to HEAD? They may not be aware of the recommended pattern you mention above.

Appending styles at run-time on-demand will cause the browser to perform style recalculations. I asked a question related in a separate comment re: rAF and batching.

Reply:
The same thing which can happen with `load-themed-styles` or webpack's css loader. The intent of the stylesheets in theme is because there is a need for them to be packaged with the theme. There are a million things which can go wrong with css in general but it's not the duty of this utility to compile and validate css.